tests/api/printf.cpp

```diff
@@ -227,4 +234,35 @@ TEST_F(WithCommandQueue, PrintfMissingLengthModifier) {
     ASSERT_STREQ(printf_buffer, message);
 }
 
+void printf_callback(const char* buffer, size_t len, size_t complete,
+                     void* user_data) {
+    ASSERT_EQ(strlen(buffer), len);
+}
+
+TEST_F(WithPrintfEnabled, PrintSimple) {
+    const char message[] = "Hello World!";
+    char source[512];
+
+    sprintf(source, "kernel void test_printf() { printf(\"%s\");}", message);
+    /* Create a cl_context with a printf_callback and user specified buffer
+     * size. */
+    cl_context_properties properties[] = {
+        /* Enable a printf callback function for this context. */
+        CL_PRINTF_CALLBACK_ARM,
+        (cl_context_properties)printf_callback,
+        CL_PRINTF_BUFFERSIZE_ARM,
+        strlen(message),
```

Comment:
No need to have the buffer than small, let's make it a bit bigger than the message that will be passed.